tree_in_context/src/lib.rs

```diff
@@ -37,11 +41,54 @@ pub fn prologue() -> String {
     .join("\n")
 }
 
+// Returns a variable representing the
+fn print_with_intermediate_helper(
+    termdag: &TermDag,
+    term: Term,
+    cache: &mut HashMap<Term, String>,
+    res: &mut String,
```

Comment:
can you implement this as an `impl io::Write` instead? Instead of `x.push_str(format!(...))` you can do `write!(x, ...)`, and you also avoid a copy.